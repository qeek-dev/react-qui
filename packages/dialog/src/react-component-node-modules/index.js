import 'normalize.css'

export { default as TextDialog } from './components/Dialog/TextDialog'
export { default as PopupDialog } from './components/Dialog/PopupDialog'
export { default as DialogContainer } from './components/DialogContainer/DialogContainer'
