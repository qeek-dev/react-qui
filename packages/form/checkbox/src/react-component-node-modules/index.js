export { default as Checkbox } from './components/Checkbox/Checkbox'
export { default as CheckboxGroup } from './components/CheckboxGroup/CheckboxGroup'

