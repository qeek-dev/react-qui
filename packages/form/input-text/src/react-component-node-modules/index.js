export { default as InputText } from './components/InputText/InputText';
