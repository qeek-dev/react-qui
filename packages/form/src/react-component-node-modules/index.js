export { default as Checkbox } from './components/Checkbox/Checkbox'
export {
  default as CheckboxGroup,
} from './components/CheckboxGroup/CheckboxGroup'
export { default as Label } from './components/Label/Label'
export { default as InputText } from './components/InputText/InputText'
export { default as Qselect } from './components/Qselect/Qselect'
