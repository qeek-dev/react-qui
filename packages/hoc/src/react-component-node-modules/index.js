export { default as withDraggable } from './hoc/withDraggable';
