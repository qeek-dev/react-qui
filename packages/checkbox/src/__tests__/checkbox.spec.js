import React from 'react'
import ReactDOM from 'react-dom'
import { mount, shallow } from 'enzyme'
import { Checkbox } from '../react-component-node-modules/index'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

it('render <Checkbox> component', () => {
  const checkbox = shallow(<Checkbox>normal</Checkbox>)
  const checkboxDisabled = shallow(<Checkbox disabled>disabled</Checkbox>)
  const checkboxChecked = shallow(<Checkbox checked>checked</Checkbox>)
  const checkboxCheckedDisabled = shallow(
    <Checkbox checked disabled>
      checked disabled
    </Checkbox>,
  )
  expect(checkbox).toMatchSnapshot()
  expect(checkboxDisabled).toMatchSnapshot()
  expect(checkboxChecked).toMatchSnapshot()
  expect(checkboxCheckedDisabled).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Checkbox>Test</Checkbox>, div)
})

it('checkbox image changed after clicking label', () => {
  const checkbox = mount(<Checkbox>Test</Checkbox>)

  expect(checkbox.find('img').prop('src')).toEqual('btn_checkbox.svg')
  checkbox.find('input').simulate('click')
  expect(checkbox.find('img').prop('src')).toEqual('btn_checkbox_pressed.svg')
})

it('Checkbox text equal to children text', () => {
  const checkbox = mount(<Checkbox>normal</Checkbox>)

  expect(checkbox.find('span').text()).toEqual('normal')
})

it('renders an image with src correctly', () => {
  const checkbox = mount(<Checkbox>normal</Checkbox>)
  const checkboxDisabled = mount(<Checkbox disabled>disabled</Checkbox>)
  const checkboxChecked = mount(<Checkbox checked>checked</Checkbox>)
  const checkboxCheckedDisabled = mount(
    <Checkbox checked disabled>
      checked disabled
    </Checkbox>,
  )

  expect(checkbox.find('img').prop('src')).toEqual('btn_checkbox.svg')
  expect(checkboxDisabled.find('img').prop('src')).toEqual(
    'btn_checkbox_disable.svg',
  )
  expect(checkboxChecked.find('img').prop('src')).toEqual(
    'btn_checkbox_pressed.svg',
  )
  expect(checkboxCheckedDisabled.find('img').prop('src')).toEqual(
    'btn_checkbox_presseddisable.svg',
  )
})
