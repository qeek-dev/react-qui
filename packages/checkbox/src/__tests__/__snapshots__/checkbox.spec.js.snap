// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Checkbox> component 1`] = `
<Checkbox
  checked={false}
  disabled={false}
  handleClick={[Function]}
  theme="light"
>
  normal
</Checkbox>
`;

exports[`render <Checkbox> component 2`] = `
<Checkbox
  checked={false}
  disabled={true}
  handleClick={[Function]}
  theme="light"
>
  disabled
</Checkbox>
`;

exports[`render <Checkbox> component 3`] = `
<Checkbox
  checked={true}
  disabled={false}
  handleClick={[Function]}
  theme="light"
>
  checked
</Checkbox>
`;

exports[`render <Checkbox> component 4`] = `
<Checkbox
  checked={true}
  disabled={true}
  handleClick={[Function]}
  theme="light"
>
  checked disabled
</Checkbox>
`;
